spec-tests/src/main/java/org/candlepin/spec/bootstrap/data/util/ExportUtil.java

```diff
@@ -0,0 +1,206 @@
+     */
+    public static <T> T deserializeJsonFile(ZipFile manifest, String path, TypeReference<T> typeRef)
+        throws IOException {
+        if (manifest == null) {
+            throw new IllegalArgumentException("Manifest is null");
+        }
+
+        if (path == null || path.isEmpty()) {
+            throw new IllegalArgumentException("Path is null or empty");
+        }
+
+        if (typeRef == null) {
+            throw new IllegalArgumentException("TypeRef is null or empty");
+        }
+
+        ZipEntry entry = manifest.getEntry(path);
+        try (InputStream istream = manifest.getInputStream(entry)) {
+            return (entry != null && !entry.isDirectory()) ?
+                ApiClient.MAPPER.readValue(istream, typeRef) :
+                null;
+        }
+    }
+
+    /**
+     * Deserializes a json file contained in the provided manifest {@ZipFile}.
+     *
+     * @param manifest
+     *  manifest file that contains the json file to deserialize; cannot be null.
+     *
+     * @param path
+     *  the fully qualified path of the json file to deserialize; cannot be null or empty.
+     *
+     * @param type
+     *  class used for deserializing the json file; cannot be null.
+     *
+     * @throws IllegalArgumentException
+     *  if the manifest or typeRef is null, or if the path is null or empty.
+     *
+     * @throws IOException
+     *  if unable to deserialize the json file.
+     *
+     * @return
+     *  deserialized json file or null if the file does not exist in the manifest.
+     */
+    public static <T> T deserializeJsonFile(ZipFile manifest, String path, Class<T> type)
+        throws IOException {
+        if (manifest == null) {
+            throw new IllegalArgumentException("Manifest is null");
+        }
+
+        if (path == null || path.isEmpty()) {
+            throw new IllegalArgumentException("Path is null or empty");
+        }
+
+        if (type == null) {
+            throw new IllegalArgumentException("Type is null or empty");
+        }
+
+        ZipEntry entry = manifest.getEntry(path);
+        try (InputStream istream = manifest.getInputStream(entry)) {
+            return (entry != null && !entry.isDirectory()) ?
+                ApiClient.MAPPER.readValue(istream, type) :
+                null;
+        }
+    }
+
+    private static boolean isEntryInDirectory(String directoryPath, ZipEntry entry) {
+        String fullyQualifiedPath = entry.getName();
+        if (!directoryPath.endsWith("/")) {
+            directoryPath = directoryPath + "/";
+        }
+
+        if (!fullyQualifiedPath.startsWith(directoryPath)) {
+            return false;
+        }
+
+        String remainingPath = fullyQualifiedPath
+            .substring(directoryPath.length(), fullyQualifiedPath.length());
+
+        // Return false if the file is in a sub-directory.
+        return !remainingPath.contains("/");
+    }
+
+}
```

Comment:
I'm not sure if there is a better way of validating that the ZipEntry is not in a sub directory.

Reply:
Can you provide some context? At the surface, the fetching of the ZipEntry in itself should be enough to verify its path:

```
ZipFile file = ...
ZipEntry entry = file.getEntry("/path/to/file.txt"); // entry.txt is in /path/to if this returns a ZipEntry
```

Reply:
This is for the getFilesInDirectory method and so I was wanting to return all files in the directory path that is provided. Alternatively we could get rid of the getFilesInDirectory method an just verify that specific expected files exists by fetching the ZipEntries like you said, but we can't validate that an unexpected file is not added to the directory. Maybe that isn't much of a concern though, but I figured it was a nice to have.

The goal of the isEntryInDirectory was to prevent returning ZipEntry when the file exists in a sub directory of the expected path:
expected directory path: `/path/to/`
file in sub directory `/path/to/ugh/file.txt`

Reply:
Well, really using the export's stream and filtering on the start of the file name matching the path will give us all the files in that directory and sub directories and so we can just assert on that collection and we can tell if an unexpected file exists or an expected file does not exists! Sorry @Ceiu, I think you said this in another comment and I was just over thinking things.  I removed the isEntryInDirectory method.

```
export.stream()
  .filter(entry -> !entry.isDirectory() && entry.getName().startsWith(path))
  .collect(Collectors.toList());
```